File: ConversorBinario/app/src/main/java/com/example/conversorbinario/MainActivity.java
package com.example.conversorbinario;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.TextView;

import org.w3c.dom.Text;

public class MainActivity extends AppCompatActivity {
    TextView buton0;
    TextView buton1;
    TextView clear;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        buton0 = (TextView) findViewById(R.id.button3);
        buton1 = (TextView) findViewById(R.id.button2);
        clear = (TextView) findViewById((R.id.button4));
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        String binary = ((TextView) findViewById(R.id.textView2)).getText().toString();
        System.out.println(binary);
        final TextView tvBin = (TextView) findViewById(R.id.textView2);
        int decimal=Integer.parseInt(binary,2);
        final TextView tvDecimal = (TextView)findViewById(R.id.textView3);
        tvDecimal.setText(Integer.toString(decimal));
        System.out.println(decimal);
        buton0.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                (TextView) ((TextView) findViewById(R.id.textView2)).setText("0");
            }
        });
        buton1.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                tvBin.setText(tvBin+"1");
            }
        });
        clear.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                tvBin.setText("");
                tvDecimal.setText("");
            }
        });
    }
}

File: app/src/main/java/com/uninorte/volleycarlos/MainActivity.kt
package com.uninorte.volleycarlos

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import com.android.volley.Request
import com.android.volley.Response
import com.android.volley.toolbox.JsonObjectRequest
import com.android.volley.toolbox.StringRequest
import kotlinx.android.synthetic.main.activity_main.*
import org.json.JSONArray
import org.json.JSONObject

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        button.setOnClickListener{
            VolleySingleton.getInstance(this).addToRequestQueue(getJsonObjectRequest())
        }
    }

    fun getStringRequest() : StringRequest {
        val url = "https://www.google.com"

        val stringRequest = StringRequest(
            Request.Method.GET, url,
            Response.Listener<String> { response->
                textView.text = response.toString()
            },
            Response.ErrorListener {
                textView.text = "error"
            }
        )
        return stringRequest
    }

    fun getJsonObjectRequest() : JsonObjectRequest {
        val url = "https://randomuser.me/api/?results=20"

        val jsonObjectRequest = JsonObjectRequest(
            Request.Method.GET, url, null,
            Response.Listener { response->
                //parseObject(response)
                parseObjectG(response)

            },
            Response.ErrorListener {
                textView.text = "error"
            }
        )
        return jsonObjectRequest
    }
    fun parseObject(response: JSONObject){
        val jsonArrayResults : JSONArray = response.getJSONArray("results")
        val size: Int = jsonArrayResults.length()
        val i: Int =0
        for (i in 0.. size -1){
            val userObject = jsonArrayResults.getJSONObject(i)
            val gender = userObject.getString("gender")
            val nameObject = userObject.getJSONObject("name")
            val firstName = userObject.getString("first")
            Log.d("JSONParsing",gender + " " + firstName)
        }
    }
    fun parseObjectG(response: JSONObject){
        var list = RandomUser.getUser(response)
        var text : String = ""
        val i:Int =0
        val size: Int = list.size
        for (i in 0.. size -1){
            val user = list.get(i)
            Log.d("WebJson", "element " + user.name.first)
            text = text + " " + user.name.first + "\n"
        }
        textView.text = text
    }

}
